### ISSUE TITLE
Event loop is closed errors v1.2.0

### ISSUE THREAD

**Issue by itaymeir25** [mjpieters/aiolimiter#292]
Since the release of version v1.2.0 on Dec 1st, we are constantly getting errors when calling `async with rate_limiter:`, causing our code to fail.
The errors are very frequent, but not constant, meaning sometimes the code randomly succeed, but most of the times it fails for the same error.
Attached below example of the code, and the Error Traceback.

### Code Example:
```
import asyncio
from aiolimiter import AsyncLimiter
rate_limiter = AsyncLimiter(20, time_period=1)

async def handle_pull_data(account_ids: list[str]):
    await asyncio.gather(*[pull_data(account_id=account_id) for account_id in account_ids])

async def pull_data(account_id: str):
    async with rate_limiter:
        await fetch_data(account_id=account_id)
```

### Error Traceback:
```
File "/workspace/main.py", line 5, in handle_pull_data
async with rate_limiter:
File "/layers/google.python.pip/pip/lib/python3.11/site-packages/aiolimiter/leakybucket.py", line 158, in __aenter__
await self.acquire()
File "/layers/google.python.pip/pip/lib/python3.11/site-packages/aiolimiter/leakybucket.py", line 119, in acquire
self._wake_next()
File "/layers/google.python.pip/pip/lib/python3.11/site-packages/aiolimiter/leakybucket.py", line 141, in _wake_next
fut.set_result(None)
File "/layers/google.python.runtime/python/lib/python3.11/asyncio/base_events.py", line 762, in call_soon
self._check_closed()
File "/layers/google.python.runtime/python/lib/python3.11/asyncio/base_events.py", line 520, in _check_closed
raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
```

**Comment by mjpieters** [comment]
Interesting; one thing that changed is that the limiter caches a reference to the running loop.

Given that your traceback contains `/layers/google.python.runtime` I am guessing you are running this code in a Google Cloud function or Google App Engine? I suspect they do something funky with the asyncio loop and caching the loop isn't supported. If so then the futures for some of your tasks were created with the cached loop but were supposed to be created with a different loop, perhaps in a different execution context.

Could you try running the following function after importing but before starting tasks with the limiter?

```
import asyncio
from aiolimiter import AsyncLimiter

def with_fresh_loop():
    @property
    def fresh_loop(self):
        return asyncio.get_running_loop()

    prop = AsyncLimiter._loop
    if prop.fget.__module__ == AsyncLimiter.__module__:
        AsyncLimiter._loop = fresh_loop
```

It'll monkey-patch the property to always look up the loop a-fresh.

If this solves the issue I'll create a new release to remove that property.

**Comment by mjpieters** [comment]
Version 1.2.1 (just released) will now issue a `RuntimeWarning` when it detects that the loop has changed, and will attempt to clear out any futures not attached to the new loop.

If you still see the issue with 1.2.1, feel free to open a new issue.

**Comment by itaymeir25** [comment]
Yes correct, we are using Python Google Cloud Functions.
We've tried the code snippet you've suggested above, and it looks like it fixed the issue, but we're still monitoring to verify that the issue is indeed completely gone.
We'll make sure to update to version 1.2.1 and test again.

In regards to the global limiter, we need a globally shared rate limiter for our use case because of external API limitations.

Thank you for the super fast response and hotfix release!

**Comment by mjpieters** [comment]
> In regards to the global limiter, we need a globally shared rate limiter for our use case because of external API limitations.

I suspected as much; combined with the fact that the simplistic example at the start of the project documentation sort-of implied a global reference is fine, I decided to try and support that use case and only issue a runtime warning.

Do let me know if, over a decent period of time, the limits are still working for you and there are no other issues and I'll remove the warning altogether to make this an officially supported feature.

**Comment by itaymeir25** [comment]
Quick update:
Our Cloud Functions have been running with the latest version of the library (v1.2.1) for the past week, with a global rate limiter instance, and are working perfectly since the hotfix release.
We've also done a few tests to make sure the rate limit is kept correctly, even with the global instance, and it works perfectly as expected.
 
Wanted to thank you again for the super fast hotfix release.
